### PR TITLE
#191 windows port

### LINKED ISSUE #191: Windows compilation

### PR DESCRIPTION
**Pull request by valleefred**
Closes #191 

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non regression tests were added (update of algorithms)
- [x] main documentation was updated (update of input/output file, update of algorithms)
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [x] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Launcher/DYNComputeLoadVariationLauncher.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ ComputeLoadVariationLauncher::launch() {
   boost::shared_ptr<MarginCalculation> marginCalculation = multipleJobs_->getMarginCalculation();
   const boost::shared_ptr<LoadIncrease>& loadIncrease = marginCalculation->getLoadIncrease();
   std::stringstream subDir;
-  subDir << "step-" << variation_ << "/" << loadIncrease->getId();
```
[the remark this answers is not in the document]
Comment (valleefred) [reply sources/Launcher/DYNComputeLoadVariationLauncher.cpp:45]: Nothing is removed !
Path concatenation is done next line to deal with OS specific path separator.

File: README.md [hunk 2]
```diff
@@ -209,8 +299,11 @@ A. Guironnet, M. Saugier, S. Petitrenaud, F. Xavier, and P. Panciatici, “Towar
...
-* [gperftools](https://github.com/gperftools/gperftools), a collection of a high-performance multi-threaded malloc implementations distributed under the BSD license. Dyna&omega;o-algorithms is currently using the version 2.6.1.
-* [MPICH](https://www.mpich.org/), an implementation of the Message Passing Interface (MPI) standard distributed under a BSD-like license. Dyna&omega;o-algorithms currently using the version 3.4.2.
+* onLinux:
+  * [gperftools](https://github.com/gperftools/gperftools), a collection of a high-performance multi-threaded malloc implementations distributed under the BSD license. Dyna&omega;o-algorithms is currently using the version 2.6.1.
+  * [MPICH](https://www.mpich.org/), an implementation of the Message Passing Interface (MPI) standard distributed under a BSD-like license. Dyna&omega;o-algorithms currently using the version 3.4.2.
+* on Windows:
+  * [MSMPI](https://learn.microsoft.com/en-us/message-passing-interface/microsoft-mpi?redirectedfrom=MSDN), a Microsoft implementation of the Message Passing Interface standard distributed under a MIT license. Dyna&omega;o-algorithms currently using the version 10.1.2.
```
[the remark this answers is not in the document]
Comment (valleefred) [reply README.md:318]: Information about MSPI added to installation.tex
Add file msmpi/license.tex

File: documentation/licenses/msmpi/license.tex [hunk 3]
```diff
@@ -0,0 +1,32 @@
+\documentclass[a4paper, 12pt]{article}
+
+\usepackage{enumitem}
+
+\begin{document}
+
+\title{MIT License}
```
Comment (rosiereflo) [inline documentation/licenses/msmpi/license.tex:7]: this needs to be added in the  licenses_folders table dynawo_algorithms_documentation.sh, compile.sh and clean.sh

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -34,10 +34,12 @@ It provides the following possibilities:
 Dyna&omega;o-algorithms is an open-source project and as such, questions, discussions, feedbacks and more generally any form of contribution are very welcome and greatly appreciated!
 
 For further informations about contributing guidelines, please refers to the [contributing documentation](https://github.com/dynawo/.github/blob/master/CONTRIBUTING.md).
+
 ## Dyna&omega;o algorithms Distribution
 
-You can download a pre-built Dyna&omega;o-algorithms release to start testing it. Pre-built releases are available for **Linux**:
+You can download a pre-built Dyna&omega;o-algorithms release to start testing it. Pre-built releases are available for **Linux** and **Windows**:
 - [Linux](https://github.com/dynawo/dynawo-algorithms/releases/download/v1.4.0/DynawoAlgorithms_Linux_v1.4.0.zip)
+- [Windows](https://github.com/dynawo/dynawo-algorithms/releases/download/v1.4.0/DynawoAlgorithms_Windows_v1.4.0.zip)
```
[the remark this answers is not in the document]
Comment (valleefred) [reply README.md:42]: Done

File: CMakeLists.txt [hunk 5]
```diff
@@ -194,43 +206,6 @@ if(BUILD_TESTS)
   include(UnitTest)
 endif()
 
-if(BUILD_TESTS OR BUILD_TESTS_COVERAGE)
```
[the remark this answers is not in the document]
Comment (valleefred) [reply CMakeLists.txt:197]: It has been moved to [sources/Test/CMakeLists.txt](https://github.com/dynawo/dynawo-algorithms/pull/317/files/d236ef06eb761e379468b60f19c5ea47ec15e512#diff-ee7046215a20c28edc7c7d27d879d5f1fa70b5058686c7c5cf43066ebbc12c10)

File: documentation/licenses/msmpi/license.tex (hunk 3 above)
Comment (valleefred) [reply documentation/licenses/msmpi/license.tex:7]: Done

File: nrt/nrtAlgo.py [hunk 6]
```diff
@@ -64,7 +65,7 @@ def launch(self, timeout):
 
         if os.getenv("DYNAWO_ENV_DYNAWO") is None:
             print("environment variable DYNAWO_ENV_DYNAWO needs to be defined")
-            sys.exit(1)
+            return  #sys.exit(1)
```
[the remark this answers is not in the document]
Comment (valleefred) [reply nrt/nrtAlgo.py:68]: sys.exit() just exits the child process (from the Pool in dynawo/sbin/nrt/nrt.py) and the main process waits forever for it to exit !
I will move this test to the main process - main() in nrtAlgo.py - to be able to do sys.exit(1).

**Review by rosiereflo — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**